src/docfx/lib/cache/JsonDiskCache.cs

```diff
@@ -133,6 +126,17 @@ public async Task<TError[]> Save()
             });
         }
 
+        private void UpdateCache(TValue item)
+        {
+            foreach (var cacheKey in item.GetKeys())
+            {
+                if (!_cache.TryGetValue(cacheKey, out var existingValue) || HasExpired(existingValue))
```

Comment:
This would prevent overwriting the existing non-expired value for the upcoming run.
But what about the corrupted cache file on the build server? Same email could match 2 non-expired entry(1 valid and 1 invalid), shall we clean that up as well?

Reply:
We only update the entry when it is a new entry or the cache has expired, this check does exactly that. Won't trigger update when a cached item hasn't expired.

The code will auto-heal corrupted cache files, so no need for manual clean up.

Reply:
what if the corrupted entry get loaded first?

Reply:
Actually, once the cache file loaded and then save, there will be only 1 entry saved to the cache file.
Let's say the invalid entry(email with null user) saved, will that be updated?

And we can remove the filtering expired on save since they should be updated here.
https://github.com/dotnet/docfx/blob/db43f6b6162ad8d6d8120e2b9424b9212482449a/src/docfx/lib/cache/JsonDiskCache.cs#L96